src/Jenssegers/Mongodb/MongodbServiceProvider.php

```diff
@@ -33,5 +33,16 @@ public function register()
                 return new MongoConnector($this->app['db']);
             });
         });
+
+        // Support queue failer to record failed jobs
+        $this->app->singleton('queue.failer', function ($app) {
```

Comment:
luckily they have an interface there `Illuminate\Queue\Failed\FailedJobProviderInterface`, so as long as the implemented failer class can handle writing / reading with their expected data format, it will just work, i tested with Amazon SQS on `queue:failed, queue:failed-table, queue:flush, queue:forget, queue:retry, queue:work`, it works good.
but i do get your concern, will try digging more to see if there's any better way to inject 'queue.failer' more graceful